superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx

```diff
@@ -596,10 +598,32 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
     }
   }, [availableDbs]);
 
+  useEffect(() => {
```

Comment:
We don't need this useEffect, we can just key on `validationErrors` to show the alert
{
  validationErrors && <Alert/>
}

Reply:
true good point.